File: C++/2816. Double a Number Represented as a Linked List.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* cur) {
        ListNode* prev=NULL;
        ListNode* fast = cur->next;
        while(cur) {
            cur->next = prev;
            prev = cur;
            cur = fast;
            if(fast) fast = fast->next;
        }
        return prev;
    }
    ListNode* doubleIt(ListNode* head) {
        ListNode* temp = reverseList(head);
        head = temp;
        int rem=0;        
        ListNode* list = new ListNode;
        while(temp) {
            int ttal = temp->val*2+rem;
            ListNode* res = new ListNode(0);
            list->val = ttal%10;
            rem = ttal/10;
            res->val = rem;
            res->next = list;
            list = res;
            temp = temp->next;
        }
        return (rem>0)? list:list->next;
    }
};

File: JAVA/2816. Double a Number Represented as a Linked List.java
import java.math.BigInteger;
class Solution 
{
    public ListNode doubleIt(ListNode head) 
    {
        String number = "";
        while( head != null )
        {
            number += head.val;
            head = head.next;
        }
        BigInteger a = new BigInteger(number);
        BigInteger b = new BigInteger("2");
        number = String.valueOf(a.multiply(b));
        head = null;
		ListNode tail = null;
        for(int i = 0; i < number.length(); i++) 
        {
            ListNode node = new ListNode( Character.getNumericValue( number.charAt(i) ) );
            if( head == null && tail == null )
			{
			    head = tail = node;
			}
			else
			{
			    tail.next = node;
			    tail = node;
			}
        }
        return head;
    }
}
